mypy/fastparse.py

```diff
@@ -345,6 +347,15 @@ def is_no_type_check_decorator(expr: ast3.expr) -> bool:
     return False
 
 
+def find_incorrect_expression(expr: ast3.expr | None) -> ast3.expr | None:
```

Comment:
This name seems a bit too generic, maybe:
```suggestion
def find_disallowed_expression_in_annotation_scope(expr: ast3.expr | None) -> ast3.expr | None:
```
(will need more changes at the call site)